app/helpers/notifications_helper.rb

```diff
@@ -50,4 +50,11 @@ def mod_article_user(data)
     article_username = data["article"]["path"].split("/")[1]
     { "name" => article_username, "path" => "/#{article_username}" }
   end
+
+  # This is used in the notification view's cache key, so that cached
+  # notification fragments can quickly be burst whenever a user
+  # adjusts their subscriptions to comment notifications
+  def subscription_status_indicator
+    @subscription_status_indicator ||= current_user.notification_subscriptions.sum(:id)
```

Comment:
Question: Why do we use sum(:id) here instead of the count?

Reply:
So, what I saw was that it's possible for a user to unsubscribe and re-subscribe (or subscribe to thread A and unsubscribe to thread B). In those sorts of scenarios, the count would remain the same, but we would _want_ the cache miss. `sum` is better able to handle scenarios where any `-` is balanced by a `+`. I _think_ we have elsewhere tended to use `last_updated_at`, but that doesn't work as well here, where `unsubscribe` **destroys** the object (we don't want to soft-delete `subscriptions`, there are too many records for that to work).

Reply:
This makes thanks, thank you for explaining.

Reply:
I _think_ you can also use cache key here. It's combination of the query digest + result count + last_updated_at on the last titem.
`current_user.notification_subscriptions.cache_key_with_version`
ref https://www.bigbinary.com/blog/activerecord-relation-cache-key